### PR TITLE
[Storage] Swagger update for STG 94

### PR DESCRIPTION
**Pull request by tanyasethi-msft**
Datalake has no updates in swagger

- [x] The purpose of this PR is explained in this or a referenced issue.
- [x] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [x] Tests are included and/or updated for code changes.
- [x] Updates to module CHANGELOG.md are included.
- [x] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go

### CODE REVIEW

File: sdk/storage/azblob/internal/generated/autorest.md [hunk 1]
```diff
@@ -36,7 +36,7 @@ directive:
   transform: >-
     return $.
       replaceAll(`[]string{"2021-12-02"}`, `[]string{ServiceVersion}`).
```
Comment (souravgupta-msft) [inline sdk/storage/azblob/internal/generated/autorest.md:38]: Does this need to be updated?

File: sdk/storage/azdatalake/internal/generated/constants.go [hunk 2]
```diff
@@ -6,4 +6,4 @@
 
 package generated
 
-const ServiceVersion = "2024-05-04"
+const ServiceVersion = "2024-08-04"
```
Comment (souravgupta-msft) [inline sdk/storage/azdatalake/internal/generated/constants.go:9]: Auto-generated code needs to be updated for having this version in the `x-ms-version` header

Comment (tanyasethi-msft) [reply sdk/storage/azdatalake/internal/generated/constants.go:9]: generated Datalake code

File: sdk/storage/azblob/internal/generated/autorest.md (hunk 1 above)
Comment (tanyasethi-msft) [reply sdk/storage/azblob/internal/generated/autorest.md]: Removed unnecessary line of code from here.

**Review by souravgupta-msft — APPROVED** [review]

**Pull request merged**